website/docs/r/azuread_application.html.markdown

```diff
@@ -13,7 +13,7 @@ Manages an Application within Azure Active Directory.
 
 ~> **NOTE:** The Azure Active Directory resources have been split out into [a new AzureAD Provider](http://terraform.io/docs/providers/azuread/index.html) - as such the AzureAD resources within the AzureRM Provider are deprecated and will be removed in the next major version (2.0). Information on how to migrate from the existing resources to the new AzureAD Provider [can be found here](../guides/migrating-to-azuread.html).
 
--> **NOTE:** If you're authenticating using a Service Principal then it must have permissions to both `Read and write all applications` and `Sign in and read user profile` within the `Windows Azure Active Directory` API.
+-> **NOTE:** If you're authenticating using a Service Principal then it must have permissions to both `Read and write all applications` and `Sign in and read user profile` within the `Windows Azure Active Directory` API. Note that this is the Legacy Azure Active Directory API, not the new Microsoft Graph API.
```

Comment:
```suggestion
-> **NOTE:** If you're authenticating using a Service Principal then it must have permissions to both `Read and write all applications` and `Sign in and read user profile` within the legacy `Windows Azure Active Directory` API.
```

Reply:
I dont think your edit adds any more clarity. if you note in mine i used the specific names as per the msft page, which is very important in saving users time. I often feel like TF docco is too brief, too terse, this is one of those times.

![image](https://user-images.githubusercontent.com/4975878/62447264-e77b4300-b75c-11e9-90d0-0c9dfcd03706.png)

![image](https://user-images.githubusercontent.com/4975878/62447289-f661f580-b75c-11e9-8c76-65f73461bb48.png)

Reply:
🤔 How about:

```suggestion
-> **NOTE:** If you're authenticating using a Service Principal then it must have permissions to both `Read and write all applications` and `Sign in and read user profile` within the legacy `Azure Active Directory Graph` API, *not* the `Microsoft Graph` API.
```